browser/importer/extensions_import_helpers.cc

```diff
@@ -0,0 +1,93 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/importer/extensions_import_helpers.h"
+
+#include <memory>
+#include <unordered_map>
+
+#include "base/files/file_util.h"
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "brave/common/importer/importer_constants.h"
+#include "components/value_store/value_store.h"
+#include "components/value_store/value_store_factory.h"
+#include "components/value_store/value_store_factory_impl.h"
+#include "extensions/browser/api/storage/value_store_util.h"
+#include "extensions/browser/extension_file_task_runner.h"
+#include "extensions/common/constants.h"
+
+namespace brave {
+namespace {
+using ExtensionStorageMap =
+    std::unordered_map<std::string, std::unique_ptr<value_store::ValueStore>>;
+
+std::vector<std::string> GetChromeExtensionsSettingsList(
+    const base::FilePath& source_profile,
+    std::vector<std::string> extensions_ids) {
+  auto store_factory_source =
+      base::MakeRefCounted<value_store::ValueStoreFactoryImpl>(source_profile);
+  std::vector<std::string> result;
+  for (const auto& id : extensions_ids) {
+    if (!store_factory_source->HasValueStore(
+            base::FilePath(extensions::kLocalExtensionSettingsDirectoryName)
+                .AppendASCII(id)))
+      continue;
+    result.push_back(id);
+  }
+  return result;
+}
+
+ExtensionStorageMap CreateStorages(
+    base::FilePath profile,
+    std::vector<std::string> ids,
+    scoped_refptr<value_store::ValueStoreFactory> factory) {
+  ExtensionStorageMap storages;
+  for (const auto& id : ids) {
+    storages[id] = extensions::value_store_util::CreateSettingsStore(
+        extensions::settings_namespace::LOCAL,
+        extensions::value_store_util::ModelType::EXTENSION, id, factory);
+  }
+  return storages;
+}
+
+}  // namespace
+
+void ImportStorages(base::FilePath source_profile,
+                    base::FilePath target_profile,
+                    std::vector<std::string> extensions_ids) {
+  DCHECK(
+      extensions::GetExtensionFileTaskRunner()->RunsTasksInCurrentSequence());
+  auto ids_with_settings =
+      GetChromeExtensionsSettingsList(source_profile, extensions_ids);
+
+  auto source_storages = CreateStorages(
+      source_profile, ids_with_settings,
+      base::MakeRefCounted<value_store::ValueStoreFactoryImpl>(source_profile));
+
+  auto target_storages = CreateStorages(
```

Comment:
what happens if user already has some extensions before importing?
Maybe it's overwritten with chrome's one.
IMO, we should respect user's existing settings.

Reply:
sure, nice catch, added it in the last commit, thanks!